setup.py

```diff
@@ -1,9 +1,69 @@
 import setuptools
 import os
 import subprocess
+import platform
 from os import path
 import sys
 
+def is_32bit():
+    return platform.architecture()[0] == '32bit'
+    
+def is_64bit():
+    return platform.architecture()[0] == '64bit'    
+
+def is_arm ():
+    return platform.machine().startswith('arm')
+
+def determine_cross_compile_string():
+    host_arch = platform.machine()
+    if (host_arch == 'AMD64' or host_arg == 'x86_64') and is_32bit() and sys.platform != 'win32':
+        return 'CMAKE_C_FLAGS=-m32'
+ 
+    return ''        
+    
+def determine_generator_string():
+    if sys.platform == 'win32':
+        vswhere_args = ['%ProgramFiles(x86)%\\Microsoft Visual Studio\\Installer\\vswhere.exe', '-legacy', '-latest', '-property', 'installationVersion']
```

Comment:
I don't think you need to specify the generator. Cmake was able to auto-pick a generator on my machine if I simply specified an architecture like so:
assuming we're cd'd into the build dir...
for 64bit on windows: `cmake .. -A x64`
for 32bit on windows: `cmake .. -A win32`

Reply:
I did it.... it didn't work, this always works so....

Reply:
I'm curious how it didn't work?

Anyway, if this unblocks us go ahead. Parsing the help output feels so dirty to me, but If we can figure out something better in the future we'll figure out something better in the future...

Reply:
I'm going to give it another go and try to dig down into it. The MSBuild runs are really slow, and if this uses ninja under the hood, that would be a much better experience

Reply:
CMake Error at CMakeLists.txt:15 (project):
  Failed to run MSBuild command:

    C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/MSBuild/15.0/Bin/MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 15.6.82.30579 for .NET Framework
    Copyright (C) Microsoft Corporation. All rights reserved.

    Build started 12/28/2018 12:02:10 PM.
    Project "C:\Users\henso\Source\aws-crt-python\deps_build\aws-c-common\CMakeFiles\3.12.1\VCTargetsPath.vcxproj" on node 1 (default targets).
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.Cpp.InvalidPlatform.Targets(22,7): error MSB8007: The Platform for project 'VCTargetsPath.vcxproj' is invalid.  Platform=' x64'. You may be seeing t
his message because you are trying to build a project without a solution file, and have specified a non-default Platform that doesn't exist for this project. [C:\Users\henso\Source\aws-crt-python\deps_build\aws-c-common\CMakeFiles\3.12.1\VCTargetsPat
h.vcxproj]
    Done Building Project "C:\Users\henso\Source\aws-crt-python\deps_build\aws-c-common\CMakeFiles\3.12.1\VCTargetsPath.vcxproj" (default targets) -- FAILED.

    Build FAILED.

    "C:\Users\henso\Source\aws-crt-python\deps_build\aws-c-common\CMakeFiles\3.12.1\VCTargetsPath.vcxproj" (default target) (1) ->
    (InvalidPlatformError target) ->
      C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.Cpp.InvalidPlatform.Targets(22,7): error MSB8007: The Platform for project 'VCTargetsPath.vcxproj' is invalid.  Platform=' x64'. You may be seeing
 this message because you are trying to build a project without a solution file, and have specified a non-default Platform that doesn't exist for this project. [C:\Users\henso\Source\aws-crt-python\deps_build\aws-c-common\CMakeFiles\3.12.1\VCTargetsP
ath.vcxproj]

        0 Warning(s)
        1 Error(s)